ztoc/ztoc_marshaler.go

```diff
@@ -18,22 +18,23 @@ package ztoc
 
 import (
 	"bytes"
-	"fmt"
 	"io"
-	"sort"
-	"time"
 
-	"github.com/awslabs/soci-snapshotter/compression"
 	ztoc_flatbuffers "github.com/awslabs/soci-snapshotter/ztoc/fbs/ztoc"
-	flatbuffers "github.com/google/flatbuffers/go"
 	"github.com/opencontainers/go-digest"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 // Marshal serializes Ztoc to its flatbuffers schema and returns a reader along with the descriptor (digest and size only).
 // If not successful, it will return an error.
 func Marshal(ztoc *Ztoc) (io.Reader, ocispec.Descriptor, error) {
-	flatbuf, err := ztocToFlatbuffer(ztoc)
+	var flatbuf []byte
+	var err error
+	if ztoc.Version == "0.9" {
+		flatbuf, err = ztocToFlatbuffer09(ztoc)
+	} else {
+		flatbuf, err = ztocToFlatbuffer(ztoc)
+	}
```

Comment:
Can we use switch here and return err in `default` case? I think that makes it more clear and won't have wrong value by accident (if happens, we'll have error).